src/cascadia/TerminalApp/AppActionHandlers.cpp

```diff
@@ -1634,10 +1635,11 @@ namespace winrt::TerminalApp::implementation
                     // we should have a randomStateString stored, if we don't then don't handle this
                     if (const auto randomStateString = Application::Current().as<TerminalApp::App>().Logic().RandomStateString(); !randomStateString.empty())
                     {
-                        ValueSet authentication{};
-                        authentication.Insert(L"url", Windows::Foundation::PropertyValue::CreateString(uriString));
-                        authentication.Insert(L"state", Windows::Foundation::PropertyValue::CreateString(randomStateString));
-                        _createAndSetAuthenticationForLMProvider(LLMProvider::GithubCopilot, authentication);
+                        Windows::Data::Json::JsonObject authValuesJson;
+                        authValuesJson.SetNamedValue(L"url", WDJ::JsonValue::CreateStringValue(uriString));
+                        authValuesJson.SetNamedValue(L"state", WDJ::JsonValue::CreateStringValue(randomStateString));
+
+                        _createAndSetAuthenticationForLMProvider(LLMProvider::GithubCopilot, authValuesJson.ToString());

```

Comment:
this is a JSON key contract between App and Provider that we should have been avoiding - what does it communicate, and why does it need to know the internal format of the auth blob?

Reply:
This is not the blob that ends up getting stored, this is the one to complete the auth flow, i.e. to hand the correct uri (which contains the necessary `code` to complete the authentication) and the random state string to the provider. The one that eventually gets stored has a format unknown to the App

Reply:
Eventually, this should be promoted to being a real function on an interface like `IOAuthAuthenticatedProvider` or similar. We don't need to do that _now_, but it IS still currently effectively a secret and undocumented contract (and the kind of thing I hate: using JSON to pass data from one part of the app to another (which is different from the 'opaque blob' thing, which uses JSON as an implementation detail for one component to save data about ITSELF))